src/minio/core/minio_client.py

```diff
@@ -81,3 +86,213 @@ async def test_connection(self) -> bool:
+
+        Args:
+            bucket_name: The name of the bucket containing the object.
+            key: The key of the object to retrieve.
+
+        Returns:
+            The content of the object as bytes.
+
+        Raises:
+            MinIOManagerError: If the object retrieval fails.
+        """
+        try:
+            async with self._get_client() as client:
+                response = await client.get_object(Bucket=bucket_name, Key=key)
+                body = await response["Body"].read()
+                logger.info(f"Got object {key} from bucket {bucket_name}")
+                return body
+        except Exception as e:
+            logger.error(f"Failed to get object {key} from bucket {bucket_name}: {e}")
+            raise MinIOManagerError(f"Object get failed: {e}") from e
+
+    async def delete_object(self, bucket_name: str, key: str) -> bool:
+        """
+        Deletes a single object from a bucket.
+
+        Args:
+            bucket_name: The name of the bucket containing the object.
+            key: The key of the object to delete.
+
+        Returns:
+            True if the object was deleted successfully.
+
+        Raises:
+            MinIOManagerError: If the object deletion fails.
+        """
+        try:
+            async with self._get_client() as client:
+                await client.delete_object(Bucket=bucket_name, Key=key)
+                logger.info(f"Deleted object {key} from bucket {bucket_name}")
+                return True
+        except Exception as e:
+            logger.error(
+                f"Failed to delete object {key} from bucket {bucket_name}: {e}"
+            )
+            raise MinIOManagerError(f"Object deletion failed: {e}") from e
+
+    async def list_objects(self, bucket_name: str, prefix: str = "") -> List[str]:
+        """
+        Lists objects in a bucket, optionally filtered by a prefix.
+
+        Args:
+            bucket_name: The name of the bucket to list objects from.
+            prefix: An optional prefix to filter the object keys.
+
+        Returns:
+            A list of object keys (strings).
+
+        Raises:
+            MinIOManagerError: If the object listing operation fails.
+        """
+        try:
+            async with self._get_client() as client:
+                paginator = client.get_paginator("list_objects_v2")
+                objects = []
+
+                async for page in paginator.paginate(Bucket=bucket_name, Prefix=prefix):
+                    if "Contents" in page:
+                        objects.extend([obj["Key"] for obj in page["Contents"]])
+
+                logger.info(
+                    f"Listed {len(objects)} objects in bucket {bucket_name} with prefix {prefix}"
+                )
+                return objects
+        except Exception as e:
+            logger.error(f"Failed to list objects in bucket {bucket_name}: {e}")
+            raise MinIOManagerError(f"Object listing failed: {e}") from e
```

Comment:
Re the exceptions, I'm not sure how you're picking which ones to throw - sometimes its a ManagerErr (which seems appropriate for non-specific errors), sometimes it's a BucketOp error (which seems appropriate for missing buckets, auth errors, etc) (although you could have a specific auth error I suppose)

Reply:
yea - I am not particularly being carefully here. I changed all errors to be BucketOperationError - I think it makes the most sense here.

Reply:
Ok, so you don't care about discriminating between different error types here?

Reply:
Since you want to throw the same error for all error types this is moot